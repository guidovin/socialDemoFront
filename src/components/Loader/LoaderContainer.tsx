import React, { useEffect, useState } from 'react'
import getApolloClient from '../../apolloClient'
import App from '../../App'

export default function WrapperApp() {
  const [client, setClient] = useState({})
  const [loading, setLoading] = useState(true)

  useEffect(() => {
    getApolloClient().then((client) => {
      setClient(client)
      setLoading(false)
    })
  }, [])
  return (<App client={client} loading={loading} />)

}