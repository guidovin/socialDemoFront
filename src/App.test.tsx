
import React from 'react';

import LoaderContainer from "./components/Loader/LoaderContainer"
import { render, prettyDOM, screen, wait } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
//simple test example, check ContentContainer.test to see a more fleshed out example with data fetching/mocking
it("Tests App Snapshot", async () => {
  await act(async ()=> {
    render(<LoaderContainer/>);
  });
  //allows for loading state to end before checking snapshot
  await wait(()=>{}, {timeout:1500});
  expect(
    prettyDOM()
  ).toMatchSnapshot()
});

it("Checks if Logo was rendered", async () => {
  await act(async ()=> {
    render(<LoaderContainer/>);
  });

  expect(
    await screen.findByText(/MySocial/i)
  ).toBeDefined()
});