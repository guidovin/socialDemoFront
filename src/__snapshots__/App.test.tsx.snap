// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests App Snapshot 1`] = `
"[36m<body>[39m
  [36m<div>[39m
    [36m<h3>[39m
      [0mLoading...[0m
    [36m</h3>[39m
  [36m</div>[39m
[36m</body>[39m"
`;
